db/middlewares/projectMemberMiddleware.ts

```diff
@@ -0,0 +1,43 @@
+import { getAnonymizedUser } from "src/core/utils/getAnonymizedUser"
+
+export default function projectMemberMiddleware(prisma) {
+  prisma.$use(async (params, next) => {
+    if (
+      params.model === "ProjectMember" &&
+      (params.action === "findMany" || params.action === "findFirst")
+    ) {
+      // Check if `deleted` is explicitly set to `undefined`
+      const hasExplicitUndefined =
+        "deleted" in params.args.where && params.args.where.deleted === undefined
+
+      if (!hasExplicitUndefined) {
+        params.args.where = {
+          ...params.args.where,
+          deleted: false, // ✅ Always filter out soft-deleted members
+        }
+
+        if (params.action === "delete") {
+          throw new Error(
+            "Direct deletion of ProjectMember is not allowed. Use soft delete instead."
+          )
+        }
+      }
+    }
```

Comment:
I get that this makes sense, coming from a soft-deletion perspective – is there no scenario where you might want to hard delete a project member directly? 

Right now, your middleware will prevent you from doing this deletion even in an admin portal that you might need in the future. If you are okay with going into the database/studio directly to do that, then I can imagine it works for now. Flagging to highlight potential downstream consequences.

Reply:
Agreed. This check should be removed from the middleware.

Reply:
[8d1e468](https://github.com/STAPLE-verse/STAPLE/pull/392/commits/8d1e468e4fe4216e67623296edb5dbf8a75b0caa)